File: dataStructures&algorithms/algorithm/leetCode/aboutArray/sortColors/sortColors.go
/**
 * @Author: shershon
 * @Description:
 * @Date: 2022/03/13 2:18 PM
 */

package main

// Leetcode-75
func sortColors(nums []int) {
	freq := make(map[int]int)
	count := len(nums)
	var i int

	for i = 0; i < count; i++ {
		freq[nums[i]]++
	}

	index := 0
	for i = 0; i < freq[0]; i++ {
		nums[index] = 0
		index++
	}
	for i = 0; i < freq[1]; i++ {
		nums[index] = 1
		index++
	}
	for i = 0; i < freq[2]; i++ {
		nums[index] = 2
		index++
	}
}

File: dataStructures&algorithms/algorithm/leetCode/aboutArray/sortColors/sortColors.php
<?php
/**
 * @date: 2022/3/13
 * @createTime: 2:14 PM
 */

// Leetcode-75
function sortColors(&$nums)
{
    $count = [0, 0, 0];

    for ($i = 0, $n = count($nums); $i < $n; $i++)
        $count[$nums[$i]]++;

    $index = 0;
    for ($i = 0; $i < $count[0]; $i++)
        $nums[$index++] = 0;
    for ($i = 0; $i < $count[1]; $i++)
        $nums[$index++] = 1;
    for ($i = 0; $i < $count[2]; $i++)
        $nums[$index++] = 2;
}